src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensRangeParams.cs

```diff
@@ -1,12 +1,16 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+#nullable enable
+
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic.Models
 {
-    internal record ProvideSemanticTokensDeltaParams : SemanticTokensDeltaParams
+    public record ProvideSemanticTokensRangeParams : SemanticTokensParams
     {
         public long RequiredHostDocumentVersion { get; set; }
+
+        public Range? Range { get; set; }
```

Comment:
I don't think this should be nullable since the LSP spec doesn't allow nullable for range on `textDocument/semanticTokens/range`.